designs/2019-additional-lint-targets/README.md

```diff
@@ -0,0 +1,133 @@
+- Start Date: 2019-05-12
+- RFC PR:
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# Configuring Additional Lint Targets with `.eslintrc`
+
+## Summary
+
+This proposal adds the ability to specify additional target files into configuration files. This enhancement will solve the pain that people have to use the `--ext` option with wanted file extensions even if they use plugins which support additional file types.
+
+## Motivation
+
+People have to use the `--ext` option or glob patterns to check wanted files even if they use plugins which support additional file types.
+
+```yml
+plugins:
+  - markdown
+  - html
+  - "@typescript-eslint"
+  - react
+  - vue
+```
+
+```bash
+# ESLint checks only `*.js`.
+eslint src docs
+
+# Needs `--ext` option
+eslint src docs --ext .js,.md,.html,.ts,.jsx,.tsx,.vue
+```
+
+However, using plugins which support additional file types is the intention that wants to check those files. The requirement of redundant CLI options is not reasonable.
+
+Per "[Related Discussions](#related-discussions)" section, this is very regularly requested; I want to configure additional target files with `.eslintrc`.
+
+## Detailed Design
+
+This proposal enhances `overrides` property of the config file.
+
+- If a config file in a directory has `overrides` property, ESLint checks the files which are matched by any of override entries (i.e., `files`/`excludedFiles` criteria) additionally in the directory.
+    - If any of `files` value of a override entry ended with `*`, ESLint ignores the override entry to avoid checking all files.
```

Comment:
What happens if someone does this?

```yml
overrides:
  files: ['*.js*'] # lint .js and .jsx
```

Following this proposal, it seems like this would not lint `jsx` files, but it seems like it might be desirable to lint `jsx` files in this case.

Maybe we could only take the override entry into account if the `file` value imposes some "restriction" on the file extension, but I'm not sure of the best way to detect that.

Reply:
It doesn't add `*.js*` files as the lint target automatically.

Dots can appear in the basename of filenames. For example, `webpack.config.js`. So I think that we should ignore that case of `['*.js*']` to avoid unintentional side effects. People can modify the glob to `["*.js", "*.jsx"]` easily.